lib/data_update_scripts/20200819025131_migrate_profile_data.rb

```diff
@@ -0,0 +1,13 @@
+module DataUpdateScripts
+  class MigrateProfileData
+    def run
+      User.find_each do |user|
+        # NOTE: no production users have profiles yet, but we want this script
+        # to be idempotent.
+        next if user.profile.present?
+
+        Profile.create(user: user, data: Profiles::ExtractData.call(user))
```

Comment:
Is there a chance that this will fail?
I know we have some weird data in some early users, I wonder if we should use `create!` to bubble up the issues, or send the failed errors somewhere

Reply:
I want to say that it can't fail, but computers, can't trust 'em. Here's what this does:

1. Take a user's attributes as hash.
2. Extract relevant keys from it.
3. Create a profile for the user with the data in a JSONB column.

There are no validations on the JSONB keys, only type coercions. What do you think?

Reply:
Let's cross fingers :D

Reply:
Given the high number of users we have, unless you deploy this during light hours there is a good chance the job gets killed before it can finish. A problem I have on my list to fix for Sidekiq so we can have long-running jobs. @citizen428 when you deploy this make sure to check Blazer to ensure this jobs completes. If it does not, ping me and I can run it in a console.

Reply:
@mstruve what is intended by light hours?

Unfortunately given the time zone difference, usually @citizen428 is offline when the bulk of the US team is operational. Given that, can the SRE team deploy this when you see fit? Otherwise we can do that a member of team OSS will ping a SRE team member tomorrow UTC afternoon and see if it's mergeable by then. 

cc @Ridhwana here as she has the same timezone as I do and she's more knowledgeable than I on the profile generalization code

Reply:
I mean when we are not deploying a lot since a deploy restarts the Sidekiq Workers so honestly, @citizen428 doing it during his hours is probably the best.

Reply:
Thanks, @mstruve, I'm deploying this not at 4:18 UTC which seems like it would be one of our lightest times (11:18pm Chicago area, past midnight in NY and 6:18am in Central Europe/South Afriace).